DevelopmentProposals/reporterColumnLabels/reporterColumnLabels.md

```diff
@@ -54,14 +54,15 @@ and forcing the user to specify an annotation for each Output may not be practic
 - The TableReporter should use a default column name that is very likely to be meaningful.
 
 ### Architecture
-The selected design involves four main changes:
+The selected design involves six main changes:
 
 1. Rename "annotation" to "alias" (e.g., `AbstractInput::getAnnotation()` becomes `AbstractInput::getAlias()`).
 2. If the user provides the optional second argument when connecting an Output to an Input, store the string as the Input's alias; otherwise, the alias remains null.
-3. Create `getShortLabel()` and `getLongLabel()` methods on Input:
-  - `getShortLabel()` will be called by ConsoleReporter and will return `isNull(alias) ? output_name : alias`.
-  - `getLongLabel()` will be called by TableReporter and will return `isNull(alias) ? full_path_name : alias`.
-4. Create `setAlias()` methods analogous to the current `getAnnotation()` methods so the user can change the alias after (but not before) connecting.
+3. Create `getReportedLabel()` on Input, which will always return the alias if one is provided:
+  - `getReportedLabel()` will be called by a Reporter and will return `isNull(alias) ? full_path_name : alias`.
```

Comment:
> That sounds specific to reporting.

"Label" is a generic term that is *not* specific to Reporters; the original design had methods called `getShortLabel()` and `getLongLabel()`. In lines 2 and 3, I indicated where `getShortLabel()` and `getLongLabel()` would be called in the implementation of the devprop; however, as we discussed, there may be other consumers that will want a label, with a preference for a short or long version. In our design meeting, it was decided that Input would handle the logic of preparing a short or long label. The new item 5 ("The ConsoleReporter has to truncate the ReportedLabel() […] a future PR can address more informative truncation approaches") moves this logic back to the ConsoleReporter, which contradicts the previous design decision.